include/TimeIntegrator.h

```diff
@@ -44,8 +44,8 @@ class TimeIntegrator
   bool simulation_proceeds();
 
   void prepare_for_time_integration();
-  void pre_realm_advance_stage1();
-  void pre_realm_advance_stage2();
+  void pre_realm_advance_stage1(size_t inonlin=0);
```

Comment:
What does `inonlin` stand for? `inOuterNonlinearIteration`?  Would you mind renaming this variable to be a little more descriptive?